homeassistant/components/refoss/coordinator.py

```diff
@@ -0,0 +1,42 @@
+"""Helper and coordinator for refoss."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from refoss_ha.controller.device import BaseDevice
+from refoss_ha.exceptions import DeviceTimeoutError
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, MAX_ERRORS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class RefossDataUpdateCoordinator(DataUpdateCoordinator[None]):
+    """Manages polling for state changes from the device."""
+
+    def __init__(self, hass: HomeAssistant, device: BaseDevice) -> None:
+        """Initialize the data update coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{DOMAIN}-{device.device_info.dev_name}",
+            update_interval=timedelta(seconds=15),
```

Comment:
Do I understand correctly, that the device can broadcast when it has an update? (Checking the bridge).

Reply:
When HA broadcast to search device ,   receive a response from the device ,  if  the device's IP address has changed ,  it can update  device's IP in HA.